src/shared/components/GridLayout/GridLayout.tsx

```diff
@@ -0,0 +1,71 @@
+import styled from '@emotion/styled'
+
+import { media } from '@/shared/theme'
+
+type ReponsivenessObject = Partial<Record<keyof typeof media, number>>
+
+export const GridLayout = styled.div`
+  display: grid;
+  grid-template-columns: repeat(12, 1fr);
+  grid-gap: 16px;
+
+  ${media.medium} {
+    grid-gap: 24px;
+  }
+`
+
+type GridItemProps = {
+  colStart?: number | 'initial' | ReponsivenessObject
+  colSpan?: number | ReponsivenessObject
+  rowStart?: number | 'initial' | ReponsivenessObject
+  rowSpan?: number | ReponsivenessObject
+}
+
+function isResponsiveObject(prop?: number | 'initial' | ReponsivenessObject): prop is ReponsivenessObject {
+  return !!prop && typeof prop !== 'number' && prop !== 'initial'
+}
+
+export const GridItem = styled.div<GridItemProps>`
+  ${media.base} {
+    ${({ colStart }) => isResponsiveObject(colStart) && `grid-column-start: ${colStart.base};`}
+    ${({ colSpan }) => isResponsiveObject(colSpan) && `grid-column-end: span ${colSpan.base};`}
+    ${({ rowStart }) => isResponsiveObject(rowStart) && `grid-row-start: ${rowStart.base};`}
+    ${({ rowSpan }) => isResponsiveObject(rowSpan) && `grid-row-end: span ${rowSpan.base};`}
+  }
```

Comment:
We can still use something to help create those. How about:

```suggestion

const createBreakpointGridRules = (breakpointKey: keyof ReponsivenessObject) => ({ colStart, colSpan, rowStart, rowSpan }: GridItemProps) => css`
  ${isResponsiveObject(colStart) && `grid-column-start: ${colStart[breakpointKey]};`}
  // ...
`

export const GridItem = styled.div<GridItemProps>`
  ${media.base} {
    ${createBreakpointGridRules('base')}
  }
```